DOC Add info about influence of sample_weights to User Guide, LogisticRegression (#27534)
scikit-learn/scikit-learn: doc/modules/linear_model.rst

.. math::
    :name: regularized-logistic-loss

>     \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i)) - (1 - y_i) \log(1 - \hat{p}(X_i))\right) + r(w).

Reviewer: ```suggestion
    \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i)) - (1 - y_i) \log(1 - \hat{p}(X_i))\right) + r(w),
```

.. math::
    :name: regularized-logistic-loss

>     \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i w)) - (1 - y_i) \log(1 - \hat{p}(X_i w))\right) + r(w).

Author: Aw, I missed that. Thanks!

>     \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i w)) - (1 - y_i) \log(1 - \hat{p}(X_i w))\right) + r(w).

Author: Hm, but it wouldn't it be wrong to add `w` either, I believe, since `w` in :math:`\hat{p}(X_i w)` (sorry, don't know how to render) in the loss function would mean the trainable weights, compared to the trained weights in the predict function. Do you think it would be correct to add `w`?

I found it confusing that even though the formula addresses to minimize `w`, `w` only appears in the regularization term.

>     \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i w)) - (1 - y_i) \log(1 - \hat{p}(X_i w))\right) + r(w).

Reviewer: Github supports some markdown, an since ~this year latex via $.

>     \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i w)) - (1 - y_i) \log(1 - \hat{p}(X_i w))\right) + r(w).

Reviewer: It's correct that the dependence on the coefficients $w$ is somewhat implicit. But for this PR, I would suggest of keeping it minimal and just add sample/class weights.

>     \min_{w} C \sum_{i=1}^n s_i \left(-y_i \log(\hat{p}(X_i w)) - (1 - y_i) \log(1 - \hat{p}(X_i w))\right) + r(w).

Author: Okay, and thanks for the hint with the syntax. :)

where :math:`{s_i}` corresponds to the weights assigned by the user to a

> specific training sample (the vector :math:`{s}` is formed by element-wise

Reviewer: ```suggestion
specific training sample (the vector :math:`s` is formed by element-wise
```

> multiplication of the class weights and sample weights). :math:`{y_i}` is the

Reviewer: From `:math:`{y_i}` it‘s a repetition what has already been said above or will be said below (the penalty). Please remove.

Author: Oh, okay, only now I understand your intention from your previous suggestion. I will remove.

> multiplication of the class weights and sample weights), :math:`{y_i}` is the
> real class of that sample (0 or 1), :math:`{\hat{p}(X_i w)}` is the predicted
> probability that :math:`{X_i}` belongs to the same class as :math:`{y_i}`, and
> :math:`{r(w)}` represents the regularization term, that is scaled by the
> regularization strength :math:`{C}`.

Reviewer: ```suggestion
multiplication of the class weights and sample weights).
```

regularization. Elastic-Net is equivalent to :math:`\ell_1` when
:math:`\rho = 1` and equivalent to :math:`\ell_2` when :math:`\rho=0`.

Note that the scale of the class weights and the sample weights will influence
the optimization problem. For instance, multiplying the sample weights by a
constant :math:`b>0` is equivalent to multiplying the (inverse) regularization

> strength `C` by :math:`b`. This is because user defined sample and class
> weights are multiplied to the coefficients and regularization penalizes large
> coefficients by adding a cost to the loss.

Reviewer: ```suggestion
strength `C` by :math:`b`.
```
I feel bad to also ask to also remove this sentence. It's just not 100% correct. From the optimization problem one sees that `C` could be drawn insight the sum and "integrated" into the weights $s_i$. Weights are never multiplied to coefficients, only to the per sample losses.

Author: Don't feel bad, @lorentzenchr. This it why I had wanted to do this issue.

regularization. Elastic-Net is equivalent to :math:`\ell_1` when
:math:`\rho = 1` and equivalent to :math:`\ell_2` when :math:`\rho=0`.

> Note that the scale of the class weights and the sample weights will influence

Reviewer: I think this is a bit misleading. It‘s not so much the optimization process, it‘s more the optimization problem itself that is changed and therefore the solution (the coefficients) changes.
And as a user, I don’t care too much about optimization, but about penalty strength.

Author: Got it, thanks.

regularization. Elastic-Net is equivalent to :math:`\ell_1` when
:math:`\rho = 1` and equivalent to :math:`\ell_2` when :math:`\rho=0`.

> Note that the scale of the class weights and the sample weights will influence
> the optimization process. This is because large user defined sample weights
> will lead to large coefficients and regularization penalizes large coefficients
> by adding a cost to the loss (even if all coefficients were enlarged
> proportionally to each other). The trained coefficients might thus slighly
> differ depending on the scale of class weights and sample weights defined by
> the user.
> 

Reviewer: ```suggestion
Note that the scale of the class weights and the sample weights will influence
the optimization process. For instance, multiplying the sample weights by a
constant :math:`b>0` is equivalent to multiplying the (inverse) regularization
strength `C` by :math:`b`.
```
